quantumblacklabs/kedro-viz
# [KED-2742] Setup ci for deployment

## Description

related ticket: https://github.com/quantumblacklabs/kedro-viz/compare/feature/KED-2829-setup-CI-for-deployment?expand=1

This is to set up the CI to trigger an automatic deployment to S3 on every push to remote. A newly pushed branch will trigger deployment to a new folder on the S3 bucket, otherwise new pushes will overwrite the previous deployed version of the branch. 

The deployed URL will be in the format of `http://kedro-viz-fe.s3-website.eu-west-2.amazonaws.com/feature/(branch name)/?data=demo`

Special thanks to @limdauto for introducing me to S3 and helping me troubleshoot the problems with this setup! 

## Development notes

I have added the new steps for deploying to S3 in the circle CI config file. 

## QA notes

The only way to QA this is to verify the successful deployment to S3 on the CI for this branch, which can be shown here: https://app.circleci.com/pipelines/github/quantumblacklabs/kedro-viz/4381/workflows/e4d497c5-4e39-4e8c-b48e-c8a419e7ff05/jobs/12621

The deployed version on S3 as a result of this deployment can be accessed here: 
http://kedro-viz-fe.s3-website.eu-west-2.amazonaws.com/feature/KED-2829-setup-CI-for-deployment/?data=demo

## Checklist


## Legal notice


- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
